src/index.ts

```diff
@@ -392,6 +402,38 @@ async function removeDockerPodmanImageStroage(): Promise<void> {
     }
 }
 
+async function checkIfManifestsExists(): Promise<boolean> {
+    const foundManifests = [];
+    const missingManifests = [];
+    // check if manifest exist in Podman's storage
+    core.debug(`🔍 Checking if the given image is manifest or not.`);
+    for (const manifest of sourceImages) {
+        const commandResult: ExecResult = await execute(
+            await getPodmanPath(),
+            [ "manifest", "exists", manifest ],
+            { ignoreReturnCode: true, group: true }
+        );
+        if (commandResult.exitCode === 0) {
+            foundManifests.push(manifest);
+        }
+        else {
+            missingManifests.push(manifest);
+        }
+    }
+
+    if (foundManifests.length > 0 && missingManifests.length > 0) {
+        throw new Error(`Manifest${missingManifests.length !== 1 ? "s" : ""} "${missingManifests.join(", ")}" `
+            + `not found in the Podman image storage.`);
+    }
+
+    if (missingManifests.length === sourceImages.length) {
```

Comment:
I find this kind of confusing.

I would do `return foundManifests.length === sourceImages.length`.